lib/rex/proto/http/request.rb

```diff
@@ -76,7 +76,7 @@ def initialize(method = 'GET', uri = '/', proto = DefaultProtocol)
   def update_cmd_parts(str)
     if (md = str.match(/^(.+?)\s+(.+?)\s+HTTP\/(.+?)\r?\n?$/i))
       self.method  = md[1]
-      self.raw_uri = URI.decode(md[2])
+      self.raw_uri = CGI.unescape(md[2])
```

Comment:
For the meterpreter session, the value of `str` was:

```
GET /lDpvTKraKaE8XjVcYvfzTASDnGgNNa1FGjqDq25ruwMs7EKq9n8wxPt-va5Izarzo1Y42Xrw2h1ObE_suZuQHC78OSC5/ HTTP/1.1'
```

Reply:
From the Ruby docs:

> **URI.unescape**
> This method is obsolete and should not be used. Instead, use CGI.unescape, URI.decode_www_form or URI.decode_www_form_component depending on your specific use case.

CGI and URI decode are not semantically equivalent, but in the above scenario that's not an issue. I don't have full context on all the scenarios this `cmd_parts` functionality would be used, before I go code diving it would be worth a second set of eyes here :eyes:

Reply:
After further discussion with the team this morning, we have confirmed that this should work fine, with a potential low to medium risk of breaking things. We will merge this in with the idea to monitor this to see if anything does indeed break.